dashboard/app/views/levels/_teacher.html.haml

```diff
@@ -11,11 +11,6 @@
       .non-scrollable-wrapper
         -# This is rendered into by react
         #teacher-panel-viewas
-        - if @user
-          .students.hide-as-student
-            %h4= t('teacher.panel.student')
-            - unless @section.nil? || @user.nil?
-              = link_to @user.name, teacher_dashboard_student_progress_url(@section, @user)
```

Comment:
deleting this student name link because 1) the page it links to is no longer going to be used, 2) consistency (images below), and 3) the list of students are rendered below this area, and this link takes up unnecessary space

re: consistency -- we recently added the teacher panel in new places (unit overview, script overview, lesson extras)
<img width="217" alt="Screen Shot 2019-04-03 at 4 03 12 PM" src="https://user-images.githubusercontent.com/9812299/55518943-b95ce980-562a-11e9-844e-0b7fa682607a.png">

so removing the "Student <link>Student Name</link>" element gives us more consistency with the above version used elsewhere